core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -1175,13 +1176,16 @@ VirtualHost build(VirtualHostBuilder template) {
                 checkState(sslContext.isServer(), "sslContextBuilder built a client SSL context.");
             }
 
+            final Builder<ShutdownSupport> builder = ImmutableList.builder();
+            builder.addAll(shutdownSupports);
+            builder.addAll(template.shutdownSupports);
```

Comment:
Note: If multiple virtual hosts use the same template, it is possible that a `ShutdownSupport` can be "closed" multiple times. I don't think there's an easy way to handle this at the moment, so I think it's OK to leave it for now.

Reply:
That's true. We might add a wrapper to call it one time.
However, I thought it's ok as it is because we are calling shutdown multiple times before this change and we don't have any problems so far.
Let me get back to this later if it makes any problem. 😉